chore(docs): update `migrating-to-v5.mdx` client component example copy

chriseling opened the pull request:
Updating docs to clarify based on speedbumps I encountered during the migration process.  Please check for correctness.

<!--
Thanks for your interest in the project. Bugs filed and PRs submitted are appreciated!

Please fill out the information below to expedite the review and (hopefully)
merge of your pull request!

**NOTE**:

- It's a good idea to open an issue first to discuss potential changes.
- Please make sure that you are _NOT_ opening a PR to fix a potential security vulnerability. Instead, please follow the [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md) to disclose the issue to us confidentially.

-->

## ☕️ Reasoning

<!-- What changes are being made? What feature/bug is being fixed here? -->

## 🧢 Checklist

- [ ] Documentation
- [ ] Tests
- [ ] Ready to be merged

## 🎫 Affected issues

<!--
Please [scout and link issues](https://github.com/nextauthjs/next-auth/issues) that might be solved by this PR. And include text like the following to close them automatically when this is merged:

Fixes: INSERT_ISSUE_LINK_HERE
-->

## 📌 Resources

- [Security guidelines](https://github.com/nextauthjs/.github/blob/main/SECURITY.md)
- [Contributing guidelines](https://github.com/nextauthjs/.github/blob/main/CONTRIBUTING.md)
- [Code of conduct](https://github.com/nextauthjs/.github/blob/main/CODE_OF_CONDUCT.md)
- [Contributing to Open Source](https://kcd.im/pull-request)

chriseling committed:
Update migrating-to-v5.mdx

Updating docs to clarify based on speedbumps I encountered during the migration process.  Please check for correctness.

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
````diff
@@ -74,8 +74,7 @@ Some things to note about the new configuration:
 ```ts filename="app/api/auth/[...nextauth]/route.ts"
-import { handlers } from "@/auth"
-export const { GET, POST } = handlers
+export { GET, POST } from '@/auth';
````
Lets revert this `handlers: { GET, POST }` change (below as well)

chriseling replied:
sgtm

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -125,11 +124,15 @@ export default async function Page() {
+const ClientComponent = () => {
+  const session = useSession();
+
+  return (<></>}
```
There's a tiny syntax typo here, but also lets just give it a small use-case example.

```suggestion
  return (
	  <>
	  	<p>Welcome {session.user.name}</p>
	  </>
  )
```

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -125,11 +124,15 @@ export default async function Page() {
 
 Imports from `next-auth/react` are now marked with the [`"use client"`](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive) directive. Therefore, they can be used in client components just like they were used in previous versions.
```
```suggestion
Imports from `next-auth/react` are now marked with the [`"use client"`](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive) directive. Therefore, they can be used in client components just like they were used in previous versions. Don't forget, client components that attempt to access the session will need to be wrapped in a `<SessionProvider />`.
```

ndom91 reviewed (commented):
So in general some nice additions, lets clean it up a bit and we can merge it :+1:

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

chriseling committed:
Update migrating-to-v5.mdx

Updating docs to clarify based on speedbumps I encountered during the migration process.

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

chriseling committed:
Merge branch 'chriseling-patch-1' of github.com:chriseling/next-auth into chriseling-patch-1

chriseling commented:
Fixed lint and formatting

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -34,7 +34,7 @@ npm install next-auth@beta
 
 - A single method to authenticate anywhere
-- Use `auth()` instead of `getServerSession`, `getSession`, `withAuth`, `getToken`, and `useSession` ([Read more](#authenticating-server-side))
+- Use `auth()` instead of `getServerSession`, `getSession`, `withAuth`, `getToken` ([Read more](#authenticating-server-side))
```
```suggestion
- Use `auth()` instead of `getServerSession`, `getSession`, `withAuth`, `getToken`, and `useSession` ([Read more](#authenticating-server-side))
```

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -123,13 +123,17 @@ export default async function Page() {
+const ClientComponent = () => {
+  const session = useSession();
+
+  return <p>Welcome {session.user.name}</p>
```
```suggestion
  return (
	  <SessionProvider>
	  	<p>Welcome {session?.user?.name}</p>
	  </SessionProvider>
  )
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -285,7 +300,15 @@ export const { handlers, auth } = NextAuth({
 import NextAuth from "next-auth"
+
+// Use middleware directly
 export const { auth: middleware } = NextAuth(authConfig)
```
```suggestion
// export const { auth: middleware } = NextAuth(authConfig)
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -285,7 +300,15 @@ export const { handlers, auth } = NextAuth({
 import NextAuth from "next-auth"
-export const { auth: middleware } = NextAuth(authConfig)
+
+// Use middleware directly
```
```suggestion
// 1. Use middleware directly
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -285,7 +300,15 @@ export const { handlers, auth } = NextAuth({
+// 1. Use middleware directly
+// export const { auth: middleware } = NextAuth(authConfig)
+
+// Wrapped middleware option, use one or the other but not both
```
```suggestion
// 2. Wrapped middleware option
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -285,7 +300,15 @@ export const { handlers, auth } = NextAuth({
 import authConfig from "./auth.config"
 import NextAuth from "next-auth"
-export const { auth: middleware } = NextAuth(authConfig)
+
```
```suggestion

// Use only one of the two middleware options below
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -123,13 +123,23 @@ export default async function Page() {
 <Tabs.Tab>
 
-Imports from `next-auth/react` are now marked with the [`"use client"`](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive) directive. Therefore, they can be used in client components just like they were used in previous versions.
+Imports from `next-auth/react` are now marked with the [`"use client"`](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive) directive. Therefore, they can be used in client components just like they were used in previous versions. Don't forget, client components that attempt to access the session will need to be wrapped in a `<SessionProvider />`.
```
```suggestion
Imports from `next-auth/react` are now marked with the [`"use client"`](https://nextjs.org/docs/getting-started/react-essentials#the-use-client-directive) directive. Therefore, they can be used in client components just like they were used in previous versions. Don't forget, client components that attempt to access the session via context will need to be wrapped in a `<SessionProvider />`.
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

k-taro56 commented on docs/pages/getting-started/migrating-to-v5.mdx:
```diff
@@ -123,13 +123,23 @@ export default async function Page() {
+  const session = useSession();
+
+  return (
+	  <SessionProvider>
+	  	<p>Welcome {session?.user?.name}</p>
+	  </SessionProvider>
```
Replace `\t`
```suggestion
    <SessionProvider>
      <p>Welcome {session?.user?.name}</p>
    </SessionProvider>
```

ndom91 committed:
Update docs/pages/getting-started/migrating-to-v5.mdx

Co-authored-by: Kawahara Shotaro <121674121+k-taro56@users.noreply.github.com>

ndom91 committed:
Merge branch 'main' into chriseling-patch-1

ndom91 committed:
Merge branch 'main' into chriseling-patch-1
